extensions/sources/random/random.go

```diff
@@ -99,7 +101,7 @@ func (s *randomSource) Open(ctx api.StreamContext, consumer chan<- api.SourceTup
 				continue
 			}
 			logger.Debugf("Send out data %v", next)
-			consumer <- api.NewDefaultSourceTuple(next, nil)
+			consumer <- api.NewDefaultSourceTuple(next, nil, conf.Clock.Now())
```

Comment:
I think we should get the time at the earliest, the same for all other sources.
And use conf.GetNow() should be fine to get the current time